app/models/secret.rb

```diff
@@ -20,7 +20,37 @@ def latest_public_keys account, kind, id
           all.
           map(&:value)
     end
+
+
+   # WITH expired_secrets AS (SELECT resource_id FROM secrets GROUP BY resource_id HAVING max(expires_at) < NOW()) SELECT resource_id, value AS ttl FROM annotations NATURAL JOIN expired_secrets WHERE name = 'ttl'
+
+# Album.group_and_count(:artist_id).having{count.function.* >= 10}
+# # SELECT artist_id, count(*) AS count FROM albums
+# # GROUP BY artist_id HAVING (count(*) >= 10)
+
+    def freshly_expired
+      Sequel::Model.db[<<-EOS
+        SELECT resource_id, expires_at, value AS ttl
+        FROM annotations
+        NATURAL JOIN (
```

Comment:
I wonder if this would be more efficient as a UNION

Reply:
I can't imagine how a `UNION` could even be useful here. Anyway, let's leave out the performance for later -- there isn't even an index on `expires_at` yet.

Reply:
Because there are two separate cases: expired secrets, and variables with a `ttl` but no secrets.

Reply:
This query does not handle the second case and I believe it's unlikely to be useful anyway.